internal/worker/environ/environ.go

```diff
@@ -25,6 +26,7 @@ var _ logger = struct{}{}
 // that allows clients to be informed of changes to the configuration.
 type ConfigObserver interface {
```

Comment:
`ConfigObserver` is a poor name for this interface.
Let's take the opportunity to rename to `ConfigAPI` (including changing any attributes called `Observer` etc